packages/@okta/vuepress-site/docs/reference/architecture-center/directory-coexistence/lab-tutorial-3-generic-database/index.md

```diff
@@ -0,0 +1,316 @@
+1. Okta stores a hash of the user's password.
+1. The user is successfully authenticated and redirected back to their application.
+
+<div class="full">
+
+  ![A flow diagram showing the Okta Password Inline Import Hook redirects authentication to a generic database from Okta if the password is not saved in Universal Directory.](/img/architecture/directory-coexistence/db-to-okta-password-inline-import-hook-flow-diagram.png)
+
+  <!--
+    Source image: fill-this-in db-to-okta-password-inline-import-hook-flow-diagram
+  -->
+</div>
+
+After this, no further calls will be made to the password validation application when that user signs in.
+
+To demonstrate the successful import of passwords through the Password Inline Import Hook, you:
+
+1. [Expose the password validation application to Okta](#expose-the-password-validation-application-to-okta)
+1. [Activate the password inline import hook in your web application](#activate-the-password-inline-import-hook-in-your-web-application)
+1. [Test the password inline import hook](#test-the-password-import-hook)
+
+## Expose the password validation application to Okta
+
+A password validation application is already running on port 8000 from a Docker container, but it is not exposed to the Internet. It needs to be exposed externally so Okta can access it. You will use `ngrok` to get a public URL for the locally running service.
+
+1. Open a new command/terminal window and run:
+
+   ```bash
+   ngrok http 8000
+   ```
+
+   Make note of your public URL. The public URL is the `https` **Forwarding** URL. In the following screenshot, the Forwarding URL is shown as `https://5d13-208-115-147-143.ngrok.io`.
+
+   <div class="full border">
+
+   ![ngrok CLI detail output](/img/architecture/directory-coexistence/db-to-okta-ngrok-url-for-generic-db.png)
+
+   </div>
+
+### Activate the password inline import hook in your web application
+
+The `pwimporthook-1` Docker container contains a Java application in the `/pw-import-hook/` directory. This application contains the password hook REST API call (`Post /pwhook`) and logic to validate the passwords.
+
+The password import inline hook must be set up and activated within your Okta Admin Console:
+
+1. Open the Admin Console for your org.
+1. Choose **Workflow > Inline Hooks**.
+1. Click **Add Inline Hook** and select **Password Import** from the dropdown menu.
+1. Add a name for the hook. For example, _Password Import Hook_.
+1. Set **URL** to your external service URL. Use the following syntax so that the **/pwhook** REST API is called on the external URL that ngrok generated. For example, `https://5d13-288-115-147-143.ngrok.io/pwhook`.
+1. Set **Authentication field** to **Authorization**.
+1. Set **Authentication secret** to **Basic dXNlcjp1c2UtYS1zdHJvbmctcGFzc3dvcmQ=**
+
+   > **Warning:** Don't forget the **Basic** prefix when entering the value into the **Authentication secret** field.
```

Comment:
When I first started, I think it took me 3 days to figure out you need to add "Basic" to the value.

Reply:
Hopefully the Caution box immediately below will help emphasize that. :-)